### DISCUSSION TITLE
Dropdown issues - PrimeOne-2.2.0 UI Kit for figma.

### DISCUSSION THREAD

**Discussion by Ola1131** [primefaces/community#2317] [Figma UI Kit]
Hi everyone,

I'm encountering an issue with dropdown lists Whenever I copy a component and try to change the dropdown list options, it keeps reverting back to the original list. This means I can't modify the content and always get the same options as shown in the picture below. This happens when I open the dropdown in prototype mode 

We use PrimeOne-2.2.0 UI Kit for figma. 

Could someone please advise on how to change the content in dropdowns? What might I be doing wrong? Any guidance or suggestions would be greatly appreciated!

![image](https://github.com/user-attachments/assets/a392ab4f-e9f6-4468-9568-da7b24046163) 
![image](https://github.com/user-attachments/assets/b5cc6e10-f50c-4764-8224-7b3b70603a9c)
![image](https://github.com/user-attachments/assets/80a6c2b0-8a06-403f-9d43-b68e0a3c5eca)

Thank you!

**Comment by onursenture** [comment]
Hi, 

This is not an issue but the expected behavior because Figma prototypes do not automatically inherit changes made to component instances.

If you don't want the values to change in prototype mode, you need to copy all the instances' states and make changes to these instances. Afterward, you need to manually add the necessary prototype options to these instances created for prototyping.

Related topic on the Figma forum: https://forum.figma.com/t/component-ignore-the-overrides-in-prototype-mode/4786

**Reply by jannisborgers** [reply]
This is frustrating and I suspect a Figma bug as well.

@onursenture The topic you link to states that all elements that need to remember the component’s state across interactions have to be available in all variants of the component.

If I understand this correctly, that means that the panel with the dropdown items needs to be available in all states of the dropdown, but only hidden in the ones where no panel is supposed to show.

I copied the panel group from the Open state, pasted it in all other variants that did not have it, and hid it, like shown in the example above, but it still does not work.

**Status quo** (PrimeOne 2.2.0 does not have panel items inside the variants where `State` is not `Open`):
![Bildschirmfoto 2024-08-30 um 17 35 44](https://github.com/user-attachments/assets/cd78056d-f9b0-431c-a16a-cd5ef7f35669)

**Should be** (Notice how the panel is present but hidden in variants where `State` is not `Open`):
![Bildschirmfoto 2024-08-30 um 16 59 29](https://github.com/user-attachments/assets/c69dbccd-be2f-43e6-b617-f3c7915cffd6)

In both cases, if I override `input->_input-content->Text` to `Choose file` instead of `Select a country` in my parent component, and add file names instead of country names to all instances of `panel->_inputlist-item->Item->Afghanistan` in all states of the instance, I get a file select that works perfectly fine when I manually switch between the variants in my designs, even if this is done by nesting components very deeply. But once I use prototype mode, clicking the dropdown resets the text values in completely unpredictable ways. Even after removing all interactions across the whole chain of nested components so that I only get the ones I need. Sometimes the old text values appear as soon as I interact with the dropdown (Choose a file -> List of country names), sometimes after selecting an item the panel closes and the input says "Text", only to change to "Albania" once I mouse over the input…

If component-scope variables where a thing, this would be much easier. Not being able to see what the text values of the items in a dropdown instance is means you have to switch to a state where they are visible first. But even that does not help, because they behave differently in prototype mode.
